contribute/git-commit-message.md

```diff
@@ -0,0 +1,116 @@
+# Commit Message Guidelines for OpenEBS Projects
+
+This document borrow some concepts, conventions and even text mainly from following sources, extending them in order to provide a sensible guideline for writing commit messages for OpenEBS projects.
+- Tim Pope's [article](http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html) on readable commit messages
+- Thanks to @stephenparish https://gist.github.com/stephenparish/9941e89d80e2bc58a153
+- Thanks to @abravalheri https://gist.github.com/abravalheri/34aeb7b18d61392251a2
+
+These conventions are aimed at tools to automatically generate useful documentation, or by developers during debugging process.
+
+## Proposed Commit Message Format
+
+Any line of the commit message cannot be longer 80 characters! This allows the message to be easier to read on github as well as in various git tools.
+
+```
+[TICKET] <type>(<scope>): <subject> <meta>
+<BLANK LINE>
+<body>
+<BLANK LINE>
+<footer>
+```
+
+### Allowed `[TICKET]` (Optional)
+
+Subject line may be prefixed for continuous integration purposes
+ or better project management with a ticket id. The ticket id could 
+ a Github Issue, Rally Id, JIRA Id, etc.,  For example,
+ if you use Rally to track your development, the subject could be
+ "[TA-1234] test(mayactl): add unit tests for cstor volume list"
+
+### Allowed `<type>`
+* **feat**: A new feature
+* **fix**: A bug fix
+* **docs**: Documentation only changes
+* **style**: Changes that do not affect the meaning of the code 
+  (white-space, formatting, missing semi-colons, etc)
+* **refactor**: A code change that neither fixes a bug nor adds a feature
+* **perf**: A code change that improves performance
+* **test**: Adding missing tests
+* **chore**: Changes to the build process or auxiliary tools and libraries
+   such as documentation generation
+
+### Allowed `<scope>`
+Scope could be anything specifying impacted module/package.
+For example: when committing to openebs/maya repo, the scope can be
+- components : mayactl, m-apiserver, spc-watcher, cast, install, util, etc.
+- generic    : compile, travis-ci, etc.
+
+### `<subject>` text
+Subject line should contains succinct description of the change. 
+
+* use imperative, present tense: “change” not “changed” nor “changes”
+* don't capitalize first letter
```

Comment:
shouldn't we capitalize first letter? Not sure - whether typo.. subject of this PR have first letter capitalized

Reply:
:-) was hoping someone will catch it